src/format.rs

```diff
@@ -14,17 +14,42 @@ use std::sync::Arc;
 const INDENT: isize = 2;
 
 pub fn pretty(src: &str) -> Result<String, crate::parse::error::ParseError> {
-    let (stripped_src, comments) = crate::parser::strip_extra(src);
-    let ast = crate::parse::parse_module(&stripped_src)?;
-    let mut formatter = Formatter::with_comments(&comments);
-    Ok(pretty_module(&ast, &mut formatter))
+    let (module, extra) = crate::parse::parse_module(&src)?;
+    let intermediate = Intermediate {
+        comments: extra
+            .comments
+            .iter()
+            .map(|span| Comment::from((span, src)))
+            .collect(),
+        doc_comments: extra
+            .doc_comments
+            .iter()
+            .map(|span| Comment::from((span, src)))
+            .collect(),
+        empty_lines: &extra.empty_lines,
+        module_comments: extra
+            .module_comments
+            .iter()
+            .map(|span| Comment::from((span, src)))
+            .collect(),
+    };
+
+    let mut formatter = Formatter::with_comments(&intermediate);
+    Ok(pretty_module(&module, &mut formatter))
+}
+
+struct Intermediate<'a> {
+    comments: Vec<Comment<'a>>,
+    doc_comments: Vec<Comment<'a>>,
+    module_comments: Vec<Comment<'a>>,
+    empty_lines: &'a [usize],
```

Comment:
This `Intermediate` struct is so I don't have to change the internals of `Formatter` which seems like it would have made this PR noisy and I didn't read `Formatter` enough to know if that was the right direction. Thoughts?

Reply:
Yeah looks good. Can always optimise later if needed